app/src/main/java/com/nio/SocketChannelWrapper.java

```diff
@@ -0,0 +1,15 @@
+package com.nio;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.channels.Selector;
+
+public interface SocketChannelWrapper {
```

Comment:
ISP is violated here. What if I will need to configure a server that should only read event and could not write?